packages/core/src/components/forms/_control-group.scss

```diff
@@ -76,69 +77,136 @@ Styleguide components.forms.control-group
     flex: 0 0 auto;
   }
 
+  // similarly to button groups, elements in control groups are stacked in a
+  // very particular order for best visual results. in each level of selector
+  // specificity, we define disabled styles last so that they override all other
+  // equally-specific styles (e.g. we don't want mouse interactions or focus
+  // changes to change the z-index of a disabled element).
```

Comment:
does it make sense to use `!important` for disabled styles like @leebyp is doing in #565?